pytorch_lightning/overrides/data_parallel.py

```diff
@@ -45,133 +40,103 @@ def _find_tensors(obj):  # pragma: no-cover
-
-    def forward(self, *inputs, **kwargs):
-        if not self.device_ids:
-            return self.module(*inputs, **kwargs)
-
-        for t in chain(self.module.parameters(), self.module.buffers()):
-            if t.device != self.src_device_obj:
-                raise RuntimeError("module must have its parameters and buffers "
-                                   "on device {} (device_ids[0]) but found one of "
-                                   "them on device: {}".format(self.src_device_obj, t.device))
-
-        inputs, kwargs = self.scatter(inputs, kwargs, self.device_ids)
-
-        if len(self.device_ids) == 1:
 
-            running_stage = self.module.running_stage
-
-            if running_stage == RunningStage.TRAINING:
-                return self.module.training_step(*inputs[0], **kwargs[0])
-
-            elif running_stage == RunningStage.TESTING:
-                return self.module.test_step(*inputs[0], **kwargs[0])
-
-            elif running_stage == RunningStage.EVALUATING:
-                return self.module.validation_step(*inputs[0], **kwargs[0])
-
-            else:
-                return self.module.predict(*inputs[0], **kwargs[0])
-
-        replicas = self.replicate(self.module, self.device_ids[:len(inputs)])
-        outputs = self.parallel_apply(replicas, inputs, kwargs)
-
-        if isinstance(outputs[0], Result):
-            outputs = self.__gather_structured_result(outputs)
-        else:
-            outputs = self.gather(outputs)
-        return outputs
+    def __init__(self, module: LightningModule, *args, **kwargs):
+        warnings.warn(
+            "The usage of `LightningDataParallel` is deprecated since v1.2 and will be removed in v1.4."
+            " From now on we recommend to directly subclass `torch.nn.parallel.DataParallel`.",
+            DeprecationWarning
+        )
+        super().__init__(LightningParallelModule(module), *args, **kwargs)
 
-    def __gather_structured_result(self, outputs):
-        prototype_output = outputs[0]
-        original_class = prototype_output.__class__
-        outputs = [dict(x) for x in outputs]
 
-        # remove all the meta info
-        meta = outputs[0]['meta']
-        for i, output in enumerate(outputs):
-            del output['meta']
+class LightningDistributedDataParallel(DistributedDataParallel):
 
-        outputs = self.gather(outputs)
+    def __init__(self, module: LightningModule, *args, **kwargs):
+        warnings.warn(
+            "The usage of `LightningDistributedDataParallel` is deprecated since v1.2 and will be removed in v1.4."
+            " From now on we recommend to directly subclass `torch.nn.parallel.DistributedDataParallel`.",
+            DeprecationWarning
+        )
+        super().__init__(LightningDistributedModule(module), *args, **kwargs)
 
-        result = original_class()
 
-        result.update(outputs)
-        result['meta'] = meta
-        return result
+class LightningModuleWrapperBase(torch.nn.Module):
```

Comment:
Is this module intended to be private, used only for the data parallel overrides? If so, should we denote that by naming it `_LightningModuleWrapperBase`?

Reply:
yes, I think that makes sense